src/btree/bt_sync_obsolete.c

```diff
@@ -429,8 +455,13 @@ __checkpoint_cleanup_walk_btree(WT_SESSION_IMPL *session, WT_ITEM *uri)
     while ((ret = __wt_tree_walk_custom_skip(
               session, &ref, __checkpoint_cleanup_page_skip, NULL, flags)) == 0 &&
       ref != NULL) {
-        if (F_ISSET(ref, WT_REF_FLAG_INTERNAL))
+        if (F_ISSET(ref, WT_REF_FLAG_INTERNAL)) {
             WT_WITH_PAGE_INDEX(session, ret = __checkpoint_cleanup_obsolete_cleanup(session, ref));
+        } else {
+            WT_ENTER_GENERATION(session, WT_GEN_SPLIT);
```

Comment:
Did you face an issue which is why you had to take a gen split here?

Reply:
No. We need to take the WT_GEN_SPLIT when operating with `ref->addr`. Before this code, it is acquired as part of the `WT_WITH_PAGE_INDEX` macro.